src/main/java/com/commercetools/sync/products/ProductSync.java

```diff
@@ -324,26 +325,17 @@ public ProductSync(@Nonnull final ProductSyncOptions productSyncOptions) {
         return productReferenceResolver
             .resolveReferences(newProductDraft)
             .thenCompose(resolvedDraft -> {
-
                 final Product oldProduct = oldProductMap.get(newProductDraft.getKey());
-
                 return ofNullable(oldProduct)
                     .map(product -> fetchProductAttributesMetadataAndUpdate(oldProduct, resolvedDraft))
                     .orElseGet(() -> applyCallbackAndCreate(resolvedDraft));
-
             })
-
             .exceptionally(completionException -> {
-
-                final ReferenceResolutionException referenceResolutionException =
-                    (ReferenceResolutionException) completionException.getCause();
                 final String errorMessage = format(FAILED_TO_RESOLVE_REFERENCES, newProductDraft.getKey(),
```

Comment:
`fetchProductAttributesMetadataAndUpdate` and `applyCallbackAndCreate` might throw an exception which is not related to ReferenceResolution? It means that this message would be wrong/confusing.

Reply:
Since our approach now is to pass the raw exception to handleError(..) no matter which type it is, the program would not distinguish exception type inside exceptionally(..). How about re-defining the error message content not to be resolve references specific?